app/Pipfile

```diff
@@ -8,13 +8,13 @@ pre-commit = "*"
 coverage = "*"
 
 [packages]
-django = "*"
-django-bootstrap4 = "*"
-django-colorfield = "*"
-daphne = "*"
-django-mathfilters = "*"
-django-crontab = "*"
-python-dateutil = "*"
+daphne = "==3.0.1"
+django = "==3.1.14"
+django-bootstrap4 = "==2.3.1"
+django-colorfield = "==0.4.1"
+django-crontab = "==0.7.1"
+django-mathfilters = "==1.0.0"
+python-dateutil = "==2.8.1"
 
 [requires]
 python_version = "3.8"
```

Comment:
Why are all dependencies versions locked in this file? Isn't that the role of `Pipfile.lock`?

By reading your commit message, it seems you want to prevent Django from being updated to a new major release, thus breaking things. In that case, I suppose we could introduce a condition, something like `django = "< 4"` (I am not sure about the syntax, though). But I don't understand the purpose of locking *all* dependency versions here.

Reply:
I locked the version for the various packages because the first time I started the project they were auto-updated to the latest versions (ignoring the `Pipfile.lock`) and it crashed because of the upgrade of `Django` to version `4.0`

I actually don't know which dependencies are safe to upgrade and which not, so I locked them all to the working versions.

Reply:
@panz3r when it crashed, which command did you run? pipenv install? 
command pipenv sync should install versions from lock file